File: shoear-web/src/features/auth/authService.js
import { apiPost, apiGet, apiPut, apiUpload, getToken } from '../../api/client';

// REAL login against the PHP API (POST /auth/login).
// `identifier` may be either an email or a username.
// Returns { token, user } on success, or throws with the server's error message.
export function login(identifier, password) {
  return apiPost('/auth/login', { identifier, password });
}

// Logout is client-side: just discard the saved token + user.
export function logout() {
  localStorage.removeItem('token');
  localStorage.removeItem('user');
}

// Start "forgot password" (POST /auth/forgot-password). Emails a reset code if
// the address has an account. Always resolves with a generic { message } —
// never reveals whether the email is registered.
export function forgotPassword(email) {
  return apiPost('/auth/forgot-password', { email });
}

// Verify a reset code WITHOUT consuming it (POST /auth/reset-password/verify-code)
// — used as its own step before the new-password step. Resolves on success.
export function verifyResetCode(email, code) {
  return apiPost('/auth/reset-password/verify-code', { email, code });
}

// Finish "forgot password" (POST /auth/reset-password): verify the emailed code
// and set a new password. Resolves with a { message } on success.
export function resetPassword(email, code, newPassword) {
  return apiPost('/auth/reset-password', { email, code, newPassword });
}

// Email a 6-digit verification code to the address being registered
// (POST /auth/register/send-code). Must succeed before register() will work.
// Resolves with a { message } on success.
export function sendRegisterCode(email) {
  return apiPost('/auth/register/send-code', { email });
}

// REAL supplier registration (POST /auth/register).
// Requires the verification code emailed by sendRegisterCode().
// Creates a Pending account; resolves with a { message } on success.
export function register(data) {
  return apiPost('/auth/register', data);
}

// Live username availability check (GET /auth/username-available). Public.
// Resolves with { available, suggestion? } — used by the sign-up/profile forms.
export function checkUsername(username) {
  return apiGet(`/auth/username-available?u=${encodeURIComponent(username)}`);
}

// Upload a business document (e.g. SSM certificate) during registration.
// Public — there's no account/token yet. Resolves with { url }.
export function uploadRegistrationDoc(file) {
  const form = new FormData();
  form.append('file', file);
  return apiUpload('/uploads/registration-doc', form);
}

// The signed-in supplier's own registration application (for the resubmit form).
// Resolves with company fields, status and any rejectionReason.
export function getMyApplication() {
  return apiGet('/supplier/application', getToken());
}

// Resubmit a corrected application after a (curable) rejection. Flips the
// account back to Pending for re-review. Resolves with { status, message }.
export function resubmitApplication(data) {
  return apiPost('/supplier/application/resubmit', data, getToken());
}

// Set/update the supplier's payout bank account. Resolves with the saved fields.
export function updateBankAccount(data) {
  return apiPut('/supplier/bank-account', data, getToken());
}

// The supplier's verified business details + any open/last change request.
export function getBusinessDetails() {
  return apiGet('/supplier/business-details', getToken());
}

// Operational (pickup) address — where couriers collect orders. Logistics, not
// verified identity, so it's editable freely (no admin review).
export function updateOperationalAddress(operationalAddress) {
  return apiPut('/supplier/operational-address', { operationalAddress }, getToken());
}

// Propose changes to the verified fields (company name, SSM, SST, document).
// Goes to the admin re-approval queue; the account stays Active meanwhile.
export function submitBusinessChangeRequest(data) {
  return apiPost('/supplier/business-details/change-request', data, getToken());
}

// The signed-in user's own profile (GET /auth/me).
export function getMe() {
  return apiGet('/auth/me', getToken());
}

// Update the signed-in user's editable fields (PUT /auth/me).
export function updateMe(data) {
  return apiPut('/auth/me', data, getToken());
}

// Change own password (POST /auth/change-password). Requires the current one.
export function changePassword(currentPassword, newPassword) {
  return apiPost('/auth/change-password', { currentPassword, newPassword }, getToken());
}

File: shoear-mobile/delivery/lib/features/auth/services/auth_service.dart
import 'dart:io';

import 'package:delivery/core/api/api_client.dart';
import 'package:delivery/features/auth/models/courier_session.dart';

/// Authentication calls against the PHP API.
class AuthService {
  final ApiClient api;
  AuthService(this.api);

  /// POST /auth/login — [identifier] is the courier's email.
  Future<CourierSession> login(String identifier, String password) async {
    final data = await api.post('/auth/login', {
      'identifier': identifier,
      'password': password,
    });
    return CourierSession.fromJson(data as Map<String, dynamic>);
  }

  /// POST /auth/register/send-code — email a 6-digit verification code to the
  /// address the courier is about to register with.
  Future<void> sendRegisterCode(String email) async {
    await api.post('/auth/register/send-code', {'email': email});
  }

  /// POST /auth/register/courier — self-apply as a courier. The account is
  /// created as Pending (awaiting admin approval), so there's no auto-login;
  /// returns the server's confirmation message.
  Future<String> registerCourier({
    required String fullName,
    required String email,
    required String phoneNumber,
    required String vehicleType,
    required String vehicleBrand,
    required String vehicleModel,
    required String vehiclePlate,
    required String password,
    required String verificationCode,
    required String licenseNumber,
    required String licensePhotoUrl,
    required String licenseClass,
    required String licenseExpiry,   // YYYY-MM-DD
    required String icNumber,
    required String icPhotoUrl,
    required String dateOfBirth,     // YYYY-MM-DD
    required bool termsAccepted,
    required String avatarUrl,
  }) async {
    final data = await api.post('/auth/register/courier', {
      'fullName': fullName,
      'email': email,
      'phoneNumber': phoneNumber,
      'vehicleType':  vehicleType,
      'vehicleBrand': vehicleBrand,
      'vehicleModel': vehicleModel,
      'vehiclePlate': vehiclePlate,
      'password': password,
      'verificationCode': verificationCode,
      'licenseNumber': licenseNumber,
      'licensePhotoUrl': licensePhotoUrl,
      'licenseClass': licenseClass,
      'licenseExpiry': licenseExpiry,
      'icNumber': icNumber,
      'icPhotoUrl': icPhotoUrl,
      'dateOfBirth': dateOfBirth,
      'termsAccepted': termsAccepted,
      'avatarUrl': avatarUrl,
    });
    return (data as Map<String, dynamic>)['message']?.toString() ??
        'Registration submitted. Your account is pending admin approval.';
  }

  /// POST /uploads/registration-doc — public (pre-login) upload for the courier's
  /// licence / IC / profile photo during registration. Returns the stored URL.
  Future<String> uploadRegistrationDoc(File file) async {
    final data = await api.uploadFile('/uploads/registration-doc', file) as Map<String, dynamic>;
    return data['url']?.toString() ?? '';
  }

  /// POST /auth/forgot-password — email a 6-digit reset code.
  Future<void> forgotPassword(String email) async {
    await api.post('/auth/forgot-password', {'email': email});
  }

  /// POST /auth/reset-password/verify-code — check the code without consuming it.
  Future<void> verifyResetCode(String email, String code) async {
    await api.post('/auth/reset-password/verify-code', {'email': email, 'code': code});
  }

  /// POST /auth/reset-password — verify the code and set a new password.
  Future<void> resetPassword(String email, String code, String newPassword) async {
    await api.post('/auth/reset-password', {'email': email, 'code': code, 'newPassword': newPassword});
  }
}
